File: types/blog.ts
export interface Blog {
  type: 'zenn' | 'qiita' | 'original';
  title: string;
  url?: string;
  slug?: string;
  date: string;
  tags: string[];
  description: string;
  published: boolean;
}

export interface ZennFeed {
  rss: {
    channel: {
      item: {
        title: string;
        link: string;
        pubDate: string;
        category: string[];
        description: string;
      }[];
    };
  };
}

export type QiitaAPI = {
  title: string;
  url: string;
  created_at: string;
  tags: {
    name: string;
  }[];
  body: string;
}[];

export interface OriginalBlogFrontMatter {
  title: string;
  date: string;
  tags: string;
  description: string;
  published: boolean;
}

export interface BlogDetail extends Blog {
  content: string;
}

File: server/entity/blog.go
package entity

import (
	"time"
)

type Blog struct {
	ID          int32      `json:"id"`
	Title       string     `json:"title"`
	Slug        string     `json:"slug"`
	Description string     `json:"description"`
	Content     string     `json:"content"`
	Tags        []*Tag     `json:"tags"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	PublishedAt *time.Time `json:"published_at"`
}
